### PR TITLE
Add Docker image build & push and testing workflows

### PR DESCRIPTION
**Pull request by JFer11**
### Description

This pull request adds two GitHub Actions workflows to the project. The first workflow, "Build and Push Docker Image," sets up Docker Buildx and builds a Docker image of the project before pushing it to Docker Hub. The second workflow, "Tests," runs tests on the project whenever changes are pushed to the repository. These workflows will improve the development and testing process for the project and make it easier to deploy to Docker Hub.

### CODE REVIEW

**Comment by JFer11** [comment]
I need to push this directly to main to test if it works

**Review by jstaricco10 — APPROVED** [review]

File: .github/workflows/build-and-push.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Build and Push Docker Image
+on:
+  push:
+    branches:
+      - main
```
Comment (jopealgorta) [inline .github/workflows/build-and-push.yml:5]: Queremos hacer esto en develop tmb no? Onda, a lo que vamos a tener dos ambientes la idea seria que cuando se pushee a develop se deployee automaticamente todo. Eso seria nuestro staging, no?

Comment (jstaricco10) [reply .github/workflows/build-and-push.yml:5]: si, de acuerdo

Comment (JFer11) [reply .github/workflows/build-and-push.yml:5]: Develop tenemos en el ambiente local con minikube. Staging no vamos a tener. Es develop en ambiente local, y produccion. Nos va a ahorrar tiempo, no tenemos tanto.

Comment (jopealgorta) [reply .github/workflows/build-and-push.yml:5]: Y bueno, vpi.

**Pull request merged**